src/Uno.UI/UI/Xaml/UIElement.cs

```diff
@@ -844,7 +844,30 @@ private static void InnerUpdateLayout(UIElement root)
 
 		internal void ApplyClip()
 		{
-#if __CROSSRUNTIME__
+#if __SKIA__
+			// On Skia specifically, we separate the two types of clipping.
+			// First, from Clip DP (handled in this code path)
+			// That clipping propagates to Visual.Clip through ApplyNativeClip.
+			// Second is clipping calculated from FrameworkElement.GetClipRect during arrange.
+			// That clipping propagates to ViewBox during arrange.
```

Comment:
This comment adds points to the idea to move corner radius directly in the `CompositionViewBox`